pkg/controller/clusterinstallation/bluegreen_utils.go

```diff
@@ -0,0 +1,206 @@
+package clusterinstallation
+
+import (
+	"context"
+	"fmt"
+
+	mattermostv1alpha1 "github.com/mattermost/mattermost-operator/pkg/apis/mattermost/v1alpha1"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	v1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// checkGreenInstallation checks the health and correctness of the k8s
+// objects that make up a GreenInstallation.
+//
+// NOTE: this is a vital health check. Every reconciliation loop should run this
+// check at the very end to ensure that everything in the installation is as it
+// should be. Over time, more types of checks should be added here as needed.
+func (r *ReconcileClusterInstallation) checkGreenInstallation(mattermost *mattermostv1alpha1.ClusterInstallation) (mattermostv1alpha1.ClusterInstallationStatus, error) {
```

Comment:
So, we definitely need to check the blue and green installation, but I believe we should be able to use one function to check any deployment type. Just like you did with most of the create functions we should be able to pass in a variable, or simply look at the config to determine how the status check should perform.

I think this is important so that we keep one area for checking the status of the cluster installation.